packages/@azure/servicebus/data-plane/lib/core/messageReceiver.ts

```diff
@@ -327,6 +327,15 @@ export class MessageReceiver extends LinkEntity {
     };
 
     this._onAmqpMessage = async (context: EventContext) => {
+      // If the receiver got closed in PeekLock mode, avoid processing the message as we
+      // cannot settle the message.
+      if (
+        this.receiveMode === ReceiveMode.peekLock &&
+        (!this._receiver || !this._receiver.isOpen())
+      ) {
+        return;
```

Comment:
Is it worth logging that we are doing this?

Reply:
Good point, yes it is.

Reply:
Done